cmd/aterm/config/config.go

```diff
@@ -1,156 +1,228 @@
+	OperationSlug  *string
+	RecordingShell *string
+}
 
-	attachStringFlag("shell", "s", "Name of the output file", t.RecordingShell, &t.RecordingShell)
-	attachStringFlag("svc", "", "Name of the output file", t.APIURL, &t.APIURL)
+func CloneConfigAsOverrides(cfg TermRecorderConfig) TermRecorderConfigOverrides {
+	strPtr := func(s string) *string { return &s }
+	return TermRecorderConfigOverrides{
+		APIURL:         strPtr(cfg.APIURL),
+		OutputDir:      strPtr(cfg.OutputDir),
+		AccessKey:      strPtr(cfg.AccessKey),
+		SecretKey:      strPtr(cfg.SecretKey),
+		OutputFileName: strPtr(cfg.OutputFileName),
+		OperationSlug:  strPtr(cfg.OperationSlug),
+		RecordingShell: strPtr(cfg.RecordingShell),
+	}
+}
 
-	flag.Parse()
+func CloneLoadedConfigAsOverrides() TermRecorderConfigOverrides {
+	return CloneConfigAsOverrides(loadedConfig)
 }
 
-// the below are small helpers to provide both short and long form flags -- not ideal, as it messes up
-// the -h flag.
+func (t *TermRecorderConfig) WriteConfigToFile(path string) error {
+	outFile, err := os.Create(path)
+	if err != nil {
+		return errors.Wrap(err, "Unable to create config file")
+	}
+	defer outFile.Close()
 
-func attachStringFlag(longName, shortName, description, defaultValue string, variable *string) {
-	if shortName != "" {
-		flag.StringVar(variable, shortName, defaultValue, description)
+	if err = yaml.NewEncoder(outFile).Encode(t); err != nil {
+		return errors.Wrap(err, "Unable to write config file")
 	}
-	flag.StringVar(variable, longName, defaultValue, description)
+	return errors.MaybeWrap(outFile.Close(), "Could not close config file")
+}
+
+func WriteConfig() error {
+	return loadedConfig.WriteConfigToFile(ATermConfigPath())
 }
 
-func attachIntFlag(longName, shortName, description string, defaultValue int64, variable *int64) {
-	if shortName != "" {
-		flag.Int64Var(variable, shortName, defaultValue, description)
+func (t *TermRecorderConfig) parseFileContent(reader io.Reader) error {
+	bytes, err := ioutil.ReadAll(reader)
+	if err != nil {
+		return errors.Wrap(err, "Unable to read config file")
+	}
+	err = yaml.Unmarshal(bytes, &t)
+	if err != nil {
+		return errors.Wrap(err, "Unable to interpret config file as YAML document")
 	}
-	flag.Int64Var(variable, longName, defaultValue, description)
+	return nil
 }
 
-func attachBoolFlag(longName, shortName, description string, defaultValue bool, variable *bool) {
-	if shortName != "" {
-		flag.BoolVar(variable, shortName, defaultValue, description)
+func (t *TermRecorderConfig) parseEnv() error {
+	err := envconfig.Process("ASHIRT_TERM_RECORDER", t)
+	if err != nil {
+		return errors.Wrap(err, "Error reading env config")
 	}
-	flag.BoolVar(variable, longName, defaultValue, description)
+	return nil
 }
 
-func attachFloatFlag(longName, shortName, description string, defaultValue float64, variable *float64) {
-	if shortName != "" {
-		flag.Float64Var(variable, shortName, defaultValue, description)
+func PrintLoadedConfig(w io.Writer) {
```

Comment:
Newlines don't seem to be properly printed here. When I get to this point after updating the config I see the following:
```
Enter a save path []: |
                       Current Configuration:
                                             	Config Version:  1
                                                                  	API Host:
         	Output Base:
                                 	Access Key:
                                                         	Secret Key:
 	Output Prefix:
                         	Operation Slug:
                                                 	Recording Shell: /usr/local/bin/zsh
           I detected problems with this configuration:
                                                        * Access Key has not been set
      * Secret Key has not been set
```

Reply:
So, for background, what's happening is that, I believe, once the pty terminal session starts, the implicit `\r` is no longer present. This is easily fixable by including it in any strings that need to be rendered post pty terminal (pre-terminal restore). This is one of the reasons why we might want to move to 2 applications, rather than a single application handling this (the other being the first-eaten-character issue that occasionally pops up). It's just a hypothesis though -- I still need to test if it will fix things.